### PR TITLE
Update instructions for client staging copy

### PR DESCRIPTION
**Pull request by verb**
Following the existing instructions results in:

```
% ./staging/copy.sh
Do not run this script directly, but via hack/update-staging-client-go.sh.
```

Also format README.md as markdown.

**What this PR does / why we need it**: Updates docs to match reality

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: 

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by verb** [comment]
@k8s-bot ok to test

File: staging/README.md [hunk 1]
```diff
@@ -1,3 +1,15 @@
...
-The staged content is copied from the main repo, i.e., k8s.io/kubernetes, with directory rearrangement and necessary rewritings. To sync the content with the latest code in your local k8s.io/kubernetes, you need to run `godep restore` in k8s root directory, then run staging/copy.sh.
-vendor/k8s.io/client-go is a symlink pointing to this staging area, so to use the packages in the staging area, you can import it as "vendor/client-go/<package-name>", as if the client were vendored. The client will be vendored from k8s.io/client-go for real after the test matrix is converted to vendor k8s components.
+This staging/src/k8s.io/client-go directory is the staging area of the client
+repo. It contains a versioned client, tools built around the client like the
+reflector, and all the client dependencies. The content will be periodically
+published to k8s.io/client-go repo.
+
+The staged content is copied from the main repo, i.e., k8s.io/kubernetes, with
+directory rearrangement and necessary rewritings. To sync the content with the
+latest code in your local k8s.io/kubernetes, you need to run `godep restore` in
```
Comment (caesarxuchao) [inline staging/README.md]: s/`godep restore`/hack/godep-restore.sh

File: staging/README.md [hunk 2]
```diff
@@ -1,3 +1,15 @@
-This staging/src/k8s.io/client-go directory is the staging area of the client repo. It contains a versioned client, tools built around the client like the reflector, and all the client dependencies. The content will be periodically published to k8s.io/client-go repo.
-The staged content is copied from the main repo, i.e., k8s.io/kubernetes, with directory rearrangement and necessary rewritings. To sync the content with the latest code in your local k8s.io/kubernetes, you need to run `godep restore` in k8s root directory, then run staging/copy.sh.
-vendor/k8s.io/client-go is a symlink pointing to this staging area, so to use the packages in the staging area, you can import it as "vendor/client-go/<package-name>", as if the client were vendored. The client will be vendored from k8s.io/client-go for real after the test matrix is converted to vendor k8s components.
+This staging/src/k8s.io/client-go directory is the staging area of the client
```
Comment (caesarxuchao) [inline staging/README.md]: This paragraph needs to be updated as well, there are many staging repos now, not just client-go.

File: staging/README.md (hunk 1 above)
Comment (verb) [inline staging/README.md]: Done

File: staging/README.md (hunk 2 above)
Comment (verb) [inline staging/README.md]: I attempted a generic rewrite of this and the first paragraph based on my (very) naive understanding of how this works. I'm happy to incorporate any specific suggestions from the experts or roll back this second commit as we've reached the limit of my knowledge in this area.

File: staging/README.md [hunk 3]
```diff
@@ -1,3 +1,15 @@
...
+
+The staged content is copied from the main repo, i.e., k8s.io/kubernetes, with
+directory rearrangement and necessary rewritings. To sync the content with the
+latest code in your local k8s.io/kubernetes, you need to run
+`hack/godep-restore.sh` in k8s root directory, then run
+`hack/update-staging-client-go.sh`.
+
+The vendor/k8s.io directory contains symlinks pointing to this staging area, so
+to use the packages in the staging area, you can import it as
+"vendor/client-go/<package-name>", as if the package were vendored. Packages
```
Comment (cblecker) [inline staging/README.md]: `"vendor/k8s.io/<package-name>"`

**Review by cblecker — CHANGES_REQUESTED** [review]

File: staging/README.md (hunk 3 above)
Comment (verb) [inline staging/README.md]: Done, thanks.

**Review by cblecker — APPROVED** [review]

File: staging/README.md [hunk 4]
```diff
@@ -1,3 +1,15 @@
-This staging/src/k8s.io/client-go directory is the staging area of the client repo. It contains a versioned client, tools built around the client like the reflector, and all the client dependencies. The content will be periodically published to k8s.io/client-go repo.
-The staged content is copied from the main repo, i.e., k8s.io/kubernetes, with directory rearrangement and necessary rewritings. To sync the content with the latest code in your local k8s.io/kubernetes, you need to run `godep restore` in k8s root directory, then run staging/copy.sh.
-vendor/k8s.io/client-go is a symlink pointing to this staging area, so to use the packages in the staging area, you can import it as "vendor/client-go/<package-name>", as if the client were vendored. The client will be vendored from k8s.io/client-go for real after the test matrix is converted to vendor k8s components.
+This directory is the staging area for packages that have been split to their
+own repository. The content here will be periodically published to respective
+top-level k8s.io repositories.
+
+The staged content is copied from the main repo, i.e., k8s.io/kubernetes, with
```
Comment (caesarxuchao) [inline staging/README.md]: Something like this:

Most code in the staging/ is authoritative, i.e., the only copy of the code. You can directly modify such code. However, the packages in staging/src/k8s.io/client-go/pkg, which are copied from pkg/. If you modify the original copy of the code, you need to run`hack/godep-restore.sh` in k8s root directory, then run `hack/update-staging-client-go.sh`. We are working towards making all code in staging/ authoritative.

**Comment by caesarxuchao** [comment]
Thanks. Two comments, otherwise lgtm.

File: staging/README.md (hunk 4 above)
Comment (verb) [inline staging/README.md]: That's much more clear, thanks.

File: staging/README.md [hunk 5]
```diff
@@ -1,3 +1,15 @@
...
+directory rearrangement and necessary rewritings. To sync the content with the
+latest code in your local k8s.io/kubernetes, you need to run
+`hack/godep-restore.sh` in k8s root directory, then run
+`hack/update-staging-client-go.sh`.
+
+The vendor/k8s.io directory contains symlinks pointing to this staging area, so
+to use the packages in the staging area, you can import it as
+"vendor/k8s.io/<package-name>", as if the package were vendored. Packages will
```
Comment (verb) [inline staging/README.md]: Done.

**Comment by verb** [comment]
@k8s-bot gce etcd3 e2e test this

**Comment by cblecker** [comment]
/lgtm

/assign @smarterclayton
for approval

**Comment by cblecker** [comment]
@k8s-bot gce etcd3 e2e test this

**Comment by caesarxuchao** [comment]
@smarterclayton could you help approve the change? thanks.

**Pull request merged**